src/Hydrators/ProjectHydrator.php

```diff
@@ -25,4 +27,30 @@ public static function getAllProjects(DatabaseConnection $db)
         }
         return $projectObjects;
     }
+
+    public static function getProjectById(DatabaseConnection $db, int $id)
+    {
+        $queryProjectAndClient = $db->prepare('SELECT projects.id, projects.name, client_id, clients.name AS client_name, clients.logo AS client_logo, deadline 
+                                                FROM projects INNER JOIN clients 
+                                                ON projects.client_id = clients.id 
+                                                WHERE projects.id=:id');
+        $queryProjectAndClient->execute(['id' => $id]);
+        $projectArray = $queryProjectAndClient->fetch();
+        if($projectArray === []) {
+            throw new InvalidProjectIdException; 
```

Comment:
Probably need a message if only because the constructor requires it

Reply:
done